ingress/watch-resource/configmap_watcher.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"log"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/registry/generic"
+	"k8s.io/kubernetes/pkg/watch"
+)
+
+// ConfigmapWatcher ...
+type ConfigmapWatcher struct {
```

Comment:
I'm not sure why I'd ever want to watch the config map directly, can you clarify the use case? the sidecar might see the config map change through watch before the app container, and the reload will be too soon. Or is the sidecar also responsible for writing the configmap into a shared volume (in which case it's duplicating kubelet functionality).